crates/core/src/pattern_compiler/snippet_compiler.rs

```diff
@@ -94,20 +94,14 @@ pub(crate) fn dynamic_snippet_from_source(
     let metavariables = split_snippet(source, context.compilation.lang);
     let mut parts = Vec::with_capacity(2 * metavariables.len() + 1);
     let mut last = 0;
-    let mut last_pos = source_range.start;
     // Reverse the iterator so we go over the variables in ascending order.
     for (byte_range, var) in metavariables.into_iter().rev() {
         parts.push(DynamicSnippetPart::String(
-            source[last as usize..byte_range.start as usize].to_string(),
+            source[last..byte_range.start].to_string(),
         ));
-        let start_pos = position_from_previous(last_pos, source, last, byte_range.start);
-        // todo: does this handle utf8 correctly?
-        last_pos = Position::new(start_pos.line, start_pos.column + var.len() as u32);
-        let range = Range::new(
-            start_pos,
-            last_pos,
-            source_range.start_byte + byte_range.start,
-            source_range.start_byte + byte_range.start + var.len() as u32,
```

Comment:
The snippet compiler has been simplified since it doesn't need to scan the source anymore to construct new `Position`s.